python/ray/data/_internal/execution/operators/map_operator_state.py

```diff
@@ -0,0 +1,506 @@
+        self._active_actors: Dict[ObjectRef[Block], ray.actor.ActorHandle] = {}
+        # The actor pool on which we are running map tasks.
+        self._actor_pool = ActorPool()
+
+    def start(self):
+        # Create the actor workers and add them to the pool.
+        ray_remote_args = self._apply_default_remote_args(self._ray_remote_args)
+        cls_ = ray.remote(**ray_remote_args)(MapWorker)
+        for _ in range(self._pool_size):
+            self._actor_pool.add_actor(cls_.remote())
+
+    def submit(
+        self,
+        transform_fn: ObjectRef[Callable[[Iterator[Block]], Iterator[Block]]],
+        input_blocks: List[ObjectRef[Block]],
+    ) -> Tuple[ObjectRef[Block], ObjectRef[BlockMetadata]]:
+        # Pick an actor from the pool.
+        actor = self._actor_pool.pick_actor()
+        # Submit the map task.
+        block, block_metadata = actor.submit.options(num_returns=2).remote(
+            transform_fn, *input_blocks
+        )
+        self._active_actors[block] = actor
+        return block, block_metadata
+
+    def task_done(self, ref: ObjectRef[Block]):
+        # Return the actor that was running the task to the pool.
+        actor = self._active_actors.pop(ref)
+        self._actor_pool.return_actor(actor)
+
+    def task_submission_done(self):
+        # Kill all idle actors in the pool, and ensure that all remaining actors in the
+        # pool will be killed as they become idle.
+        self._actor_pool.kill_idle_actors()
+        self._actor_pool.kill_future_idle_actors()
+
+    def cancellable(self):
+        # Actor tasks are not cancellable.
+        return False
+
+    @staticmethod
+    def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any]:
+        """Apply defaults to the actor creation remote args."""
+        ray_remote_args = ray_remote_args.copy()
+        if "num_cpus" not in ray_remote_args:
+            ray_remote_args["num_cpus"] = 1
+        if "scheduling_strategy" not in ray_remote_args:
+            ctx = DatasetContext.get_current()
+            if ctx.scheduling_strategy == DEFAULT_SCHEDULING_STRATEGY:
+                ray_remote_args["scheduling_strategy"] = "SPREAD"
+            else:
+                ray_remote_args["scheduling_strategy"] = ctx.scheduling_strategy
+        return ray_remote_args
+
+
+class MapWorker:
+    """An actor worker for MapOperator."""
+
+    def ready(self):
+        return "ok"
+
+    def submit(
+        self, fn: Callable[[Iterator[Block]], Iterator[Block]], *blocks: Block
+    ) -> Tuple[Block, BlockMetadata]:
+        # Coalesce all fn output blocks.
+        # TODO(Clark): Remove this coalescing once dynamic block splitting is supported
+        # for actors.
+        # yield from _map_task(fn, *blocks)
+        stats = BlockExecStats.builder()
+        builder = DelegatingBlockBuilder()
+        for block in fn(iter(blocks)):
+            builder.add_block(block)
+        block = builder.build()
```

Comment:
This will have a minor overhead increase compared to block splitting being turned off for `fn` (e.g. we might split outputs into blocks on the target block size threshold, only to concatenate them back together here), but I think this is fine since this is just a stopgap until we have real dynamic block splitting support.